### PR TITLE
feat: schematics to add iframe to a component

### PR DESCRIPTION
**Pull request by matthieu-crouzet**
## Proposed change
Refactor of the iframe schematics to be aligned with the strategy

### CODE REVIEW

**Comment by kpanot** [comment]
Ng-add should register itself to angular.json schematics collection list

File: packages/@o3r/third-party/schematics/ng-add/index.ts [hunk 1]
```diff
@@ -1,10 +1,14 @@
-import { noop } from '@angular-devkit/schematics';
 import type { Rule } from '@angular-devkit/schematics';
+import {registerPackageCollectionSchematics} from '@o3r/schematics';
```
Comment (kpanot) [inline packages/@o3r/third-party/schematics/ng-add/index.ts:2]: should be dynamic import to handle the case of non-otter project

**Review by fpaul-1A — APPROVED** [review]

File: packages/@o3r/third-party/collection.json [hunk 2]
```diff
@@ -6,6 +6,12 @@
       "factory": "./schematics/ng-add/index#ngAdd",
       "schema": "./schematics/ng-add/schema.json",
       "aliases": ["install", "i"]
+    },
+    "add-iframe-component": {
```
Comment (mrednic-1A) [inline packages/@o3r/third-party/collection.json:10]: `iframe-to-component` ? to stay aligned with the others schematics applied to components (loc-to-component, context-to-component etc)

File: packages/@o3r/third-party/package.json [hunk 3]
```diff
@@ -20,6 +20,10 @@
     "tslib": "^2.5.3",
     "uuid": "^9.0.0"
   },
+  "peerDependencies": {
+    "@angular-devkit/schematics": "~16.0.5",
```
Comment (mrednic-1A) [inline packages/@o3r/third-party/package.json:24]: these 2 can pass as optional peer deps as they are used only in schematics/generators ?

**Review by mrednic-1A — APPROVED** [review]

File: packages/@o3r/third-party/README.md [hunk 4]
```diff
@@ -18,3 +18,12 @@ ng add @o3r/third-party
...
 > **Warning**: this module requires [@o3r/core](https://www.npmjs.com/package/@o3r/core) to be installed.
+
+## Generators
+
+Otter framework provides a set of code generators based on [angular schematics](https://angular.io/guide/schematics).
+
+| Schematics            | Description                                                  | How to use                  |
+| --------------------- | ------------------------------------------------------------ | --------------------------- |
+| add                   | Include Otter third party module in a library / application. | `ng add @o3r/third-party`   |
+| iframe-to-component  | Add iframe to an Otter component                             | `ng g iframe-to-component` |
```
Comment (kpanot) [inline packages/@o3r/third-party/README.md:29]: missing space :D

**Review by kpanot — APPROVED** [review]

File: packages/@o3r/third-party/README.md (hunk 4 above)
Comment (matthieu-crouzet) [reply packages/@o3r/third-party/README.md:29]: fix in next PR

**Pull request merged**